### PR TITLE
Add pull request template for .NET SDK repository

### PR DESCRIPTION
**Pull request by cormacpayne**
- Add pull request template for .NET SDK repository as well as missing documentation
    - CONTRIBUTING.md
        - Describes how to contribute to the .NET SDK repository
    - breaking-changes.md
        - Defines what a breaking change is in the .NET SDK repository

### CODE REVIEW

**Comment by cormacpayne** [comment]
@hovsepm @shahabhijeet would you mind taking a look? Let me know if there are any gaps in the documentation I added, or anything incorrect.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    - [Breaking changes](#breaking-changes)
+    - [General guidelines](#general-guidelines)
+    - [Testing guidelines](#testing-guidelines)
+    - [SDK generation guidelines](#sdk-generation-guidelines)
+
+## Before Starting
+
+### Onboarding
+
+Make sure that your GitHub account is part of the Azure organization. [Use this page](http://aka.ms/azuregithub) to link your account.
```
Comment (hovsepm) [inline CONTRIBUTING.md]: This line is only relevant to internal teams at Microsoft. I think it is better to move it to adx-documentation-pr repo.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+### Pull Requests
+
+You can find all of the pull requests that have been opened in the [Pull Request](https://github.com/Azure/azure-sdk-for-net/pulls) section of the repository.
+
+To open your own pull request, click [here](https://github.com/Azure/azure-sdk-for-net/compare). When creating a pull request, keep the following in mind:
+- Make sure you are pointing to the fork and branch that your changes were made in
+- Choose the correct branch you want your pull request to be merged into
+    - The **AutoRest** branch is for code generated from AutoRest
+    - The **master** branch is for code generated from Hyak
```
Comment (hovsepm) [inline CONTRIBUTING.md]: please add here
**Fluent** branch is for the next generation of Azure SDK Management libraries that implement Fluent interfaces.

Comment (hovsepm) [inline CONTRIBUTING.md]: also for **master** branch please state that code there is in support mode only.

**Review by hovsepm — CHANGES_REQUESTED** [review]

**Pull request merged**